Makefile.common

```diff
@@ -116,5 +116,5 @@ TEST_COVERAGE_DIR := "test_coverage"
 
 ifneq ("$(CI)", "0")
 	BAZEL_GLOBAL_CONFIG := --output_user_root=/home/jenkins/.tidb/tmp --host_jvm_args=-XX:+UnlockExperimentalVMOptions --host_jvm_args=-XX:+UseZGC
-	BAZEL_CMD_CONFIG := --config=ci --repository_cache=/home/jenkins/.tidb/tmp
+	BAZEL_CMD_CONFIG := --config=ci
```

Comment:
If that can be done, what about `--output_user_root=/home/jenkins/.tidb/tmp`?

Reply:
All can be set in CI worker's `$HOME/.bazelrc`
- `--repository_cache` can be shared with different builds and workspaces.
- `--output_user_root` should not be shared with different workspaces.

The PR focus on optimize for shared repo cache, `--output_user_root` should be migrated in another PR.